api/admin/controller/__init__.py

```diff
@@ -372,6 +374,63 @@ def __call__(self, collection, book, path=None):
         response.set_cookie("csrf_token", csrf_token, httponly=True)
         return response
 
+class TimestampsController(AdminCirculationManagerController):
+    """Returns a dict: each key is a type of service (script, monitor, or coverage provider);
+    each value is a nested dict in which timestamps are organized by service name and then by collection ID."""
+
+    def diagnostics(self):
+        self.require_system_admin()
+        timestamps = self._db.query(Timestamp).order_by(Timestamp.start)
+        if isinstance(timestamps, ProblemDetail):
+            return timestamps
```

Comment:
It's not possible for a SQLAlchemy operation to return a ProblemDetail, so you can take this out. If there's a database problem, SQLAlchemy will raise an exception which will be handled by the app server itself.